### ISSUE TITLE
Feature: Add a list layout

### ISSUE THREAD

**Issue by yaira2** [files-community/Files#2509]
### What feature or improvement do you think would benefit Files?
The two main parts of this proposal are for a layout mode that scrolls horizontally and displays items in columns.

### Requirements

- Add new template to the GridView layout page
- Scrolls horizontally instead of vertically
- Add option to the layout flyout menu after the option for "tiles"
- Add option to right click context menu after the option for "tiles" (when clicking a blank space)
- Add shortcut to switch to layout (should follow the pattern of other layouts)
- Displays sync status if available followed by the item icon and name

### Files Version

2.3.11.0

### Windows Version

10.19043.1899

**Comment by Jaiganeshkumaran** [comment]
@yaichenbaum Instead of renaming current list view, we can name the new list view as List view compact

**Comment by Jaiganeshkumaran** [comment]
Isn't this a duplicate of #1368

**Comment by yaichenbaum** [comment]
> @yaichenbaum Instead of renaming current list view, we can name the new list view as List view compact

The current list view is really a details view.

**Comment by Jaiganeshkumaran** [comment]
@yaichenbaum Won't users get confused if we do that?

**Comment by yaichenbaum** [comment]
> @yaichenbaum Won't users get confused if we do that?

I don't think so, it would also align it with the name File Explorer uses for the details view.

**Comment by veermaharaj** [comment]
Comparison of Details View In files app, vs list view in windows explorer using the same folder with the same files.

[https://1drv.ms/u/s!AkFzMQMvMpBvhqtlWaGwNvUJuVZC2A?e=bsJiaO](url)

**Comment by Terepin** [comment]
This is the only thing that is preventing me from switching.

**Comment by dftf-stu** [comment]
I was going to log a ticket for this myself, but found this issue via the search?

Is there perhaps any milestone version or projected date when this might be added -- I have many folders which contain 100s of folders inside them, and I use "List" view in _File Explorer_ for them always.  Be really great to see this view-type get added to _Files_ too...

**Comment by yaichenbaum** [comment]
> I was going to log a ticket for this myself, but found this issue via the search.
> 
> Is there perhaps any milestone version or projected date when this might be added -- I have many folders which contain 100s of folders inside them, and I use "List" view in _File Explorer_ for them always. Be really great to see this view-type get added to _Files_ too...

We're currently focused on improving the existing layouts and app architecture to make this posible so it might be a while before we implement this.

**Comment by au-lait77** [comment]
I hope "List" comes because this is what I use 80% of the time besides "Tiles".

**Comment by dftf-stu** [comment]
@yair100 : just-wondering if there is any-update on adding the "List View" layout-option please?

It's still the biggest-thing holding-me-back from using your app, as it really-limits the total-number of folders I can see at-once.
For example, if I go into my "Music" folder, then on my 1366x768 screen I see the following with each app's window "maximised":
- **File Explorer:** 216 folders are visible.  First-folder starts "00" in its name; the last visible-folder starts "Me"
- **[File Manager](https://github.com/microsoft/winfile) (WINFILE.EXE):** 221 folders visible.  First-folder starts "00"; final-folder visible starts "Mi"
- **Files app:** just 44 folders visible (in "Grid View (small)"), going-from "00" to "Bo"

It would be great to finally have this feature added please!

**Comment by yaira2** [comment]
@dftf-stu we're continuing to make progress on the refactor and Microsoft recently announced that they are working on a new layout control for this scenario, but we still have some work to do before we can implement it.

**Comment by beedell-roke** [comment]
@yair100, do you have a link to the announcement?

**Comment by yaira2** [comment]
@rokejulianlockhart it was demoed on the most recent WinUI Community call, I think the recording is available on YouTube.

**Comment by Diogo-Rossi** [comment]
> I hope "List" comes because this is what I use 80% of the time besides "Tiles".



> This is the only thing that is preventing me from switching.

Same for me.

**Comment by CBearr** [comment]
I think that when Files finally brings a list view it will give the project a very significant user boost. Because everywhere I read comments on news or blog entries about Files, a lot of people comment that they find Files interesting but don't use it productively as an Explorer replacement because of the missing list view.

**Comment by dftf-stu** [comment]
@yaira2 Hi Yair.  Any further updates on when the "List" view might finally get added, please? 🙏🏻

This ticket was opened on 2 Dec 2020, so in just-under one-month-from-now it will be _three years_ old.  Is there any-reason why adding that one view to your app is proving so-difficult to do, given you added _every-other_ type of view years-ago? 🤔

I can only speak for myself here but... I really feel at this-point you should just consider closing this ticket and simply-say you won't be adding it.  Each time I see an update for your app in the Microsoft Store I do the update, then re-launch your app, browse into a folder and right-click hoping to see "List" appear as a layout option... but am always disappointed to see it's still not there.  By the time you may eventually add it, I (like I'd imagine many-others) will have moved to Windows 11, where File Explorer now offers tabs, along with UI improvements over the Windows 10 version, and on _11_ I see less-of-a-need to then use "Files" anyway. 🤷🏻‍♂️

**Comment by yaira2** [comment]
> Is there any-reason why adding that one view to your app is proving so-difficult to do, given you added every-other type of view years-ago?

It's a larger piece of work and we haven't had the time to do it yet. It's certainly still something we plan to add so there is no reason to close the issue.

**Comment by Fefedu973** [comment]
+1 pretty much needed at this point

**Comment by Mantej-Singh** [comment]
+1
I switch back to explorer

**Comment by yaira2** [comment]
This feature is approved for development on the todo list.

**Comment by Mantej-Singh** [comment]
Feeling better now 😊

**Comment by yaira2** [comment]
![image](https://github.com/files-community/Files/assets/39923744/6158334c-cfcb-4ba1-9b21-19e35ca8a2e6)

**Comment by Mantej-Singh** [comment]
Today is my birthday, this is a perfect gift for me. Thanks @yaira2

**Comment by yaira2** [comment]
I'm excited to share that we're actively working on a List View layout with @chingucoding leading the efforts. There is still some work to be done but you can check out the latest progress in #14489.

**Comment by yaira2** [comment]
The List View layout is now available in the preview version https://files.community/download

**Comment by Mantej-Singh** [comment]
🏃

**Comment by Mantej-Singh** [comment]
@yaira2 
Just tested it, its crashing for me. 
I'd be happy to help. Any way I provide you with my crash logs?

**Comment by Josh65-2201** [comment]
Please open a new issue using the bug report template.

**Comment by RogerFx** [comment]
Hi

I am loving how the new Files app is working and the UI. So far I have one concern:
List View does not show the complete file name

Also, my apologies if I am doing something wrong by indicating this issue in this comment. I have being reading around the forum and there is a lot to learn.

Tks, Roger

**Comment by yaira2** [comment]
@RogerFx thank you for the question, this is a limitation we have with the Grid View layout as well. We don't have a way to have dynamic item sizes without reducing performance. To work around this, we set a fixed width that works for most file names, but the downside is that sometimes it's not large enough. We'll keep looking for solutions and ways to solve this issue.
